### ISSUE TITLE
Getting started page awaits a lifecycle phase in `setInput`?

### ISSUE THREAD

**Issue by bpasero** [microsoft/vscode#183249]
(found this because this actually resulted in a error: https://github.com/microsoft/vscode/issues/183226)

Code is here:

https://github.com/microsoft/vscode/blob/bf9be2c3661005357d27b8099c54ec99e559cf31/src/vs/workbench/contrib/welcomeGettingStarted/browser/gettingStarted.ts#L308

I do not fully understand this, but instead of awaiting the phase in `setInput`, why not await it before opening the welcome page? And why does not wait at all there, what is the reason?

I think this may also be the cause of https://github.com/microsoft/vscode/issues/181660

**Comment by bhavyaus** [comment]
My bad. I wasn't aware that the ContextKeyHandler was also using the EditorInput to performs checks. 
My main goal with introducing that check in https://github.com/microsoft/vscode/pull/182947 was to check for all service ready before I attempted to fetch content for the welcome page. 

> I think this may also be the cause of https://github.com/microsoft/vscode/issues/181660 

This was a change I introduced last Friday and  https://github.com/microsoft/vscode/issues/181660 precedes it. I've already fixed the issue with: https://github.com/microsoft/vscode/pull/182796
